lib/api/client-commands/registerBasicAuth.js

```diff
@@ -0,0 +1,41 @@
+const ClientCommand = require('./_base-command.js');
+
+/**
+ * Automate the input of basic auth credentials whenever they arise.
+ *
+ * @example
+ *  this.demoTest = function (browser) {
+ *    browser.registerBasicAuth(1000, 800);
+ *    browser.url('https://page-needs-auth-credentials.com');
+ *  };
+ *
+ *
+ * @method registerBasicAuth
+ * @param {string} username
+ * @param {string} password
+ * @param {function} [callback] Optional callback function to be called when the command finishes.
+ */
+class RegisterBasicAuth extends ClientCommand {
+
+  performAction(callback) {
+    const {username, password} = this;
+
+    this.transportActions.registerAuth(
+      username,
+      password,
+      callback
+    ).catch(err => {
+      return err;
+    }).then(result => callback(result));
+
+  }
+
+  command(username, password, callback) {
+    this.username = username;
```

Comment:
we need to check the driver support for CDP and show an error when not available.

Reply:
I've used `browserName` to check whether the CDP method is supported or not